### ISSUE TITLE
iroh-ctl and iroh-one

### ISSUE THREAD

**Issue by fabricedesre** [n0-computer/iroh#224]
With the proper matching configuration on both side, `iroh-ctl` can monitor an `iroh-one` instance:

```
     Running `/media/external/dev/capyloon/iroh/target/debug/iroh-ctl --cfg config.toml status`
Process		Number	Status
gateway		1/1	Serving
p2p			1/1	Down	The service is currently unavailable
store			1/1	Down	The service is currently unavailable
```

This is not technically wrong, but still a bit misleading since no p2p or store services are expected to run in that case. How do we want to tackle that? One way would be to add a `--iroh-one` flag to `iroh-ctl` and only deal with the gateway in that case.

**Comment by ramfox** [comment]
This is an extremely timely comment.

We have been having some discussions trying to understand the role/function of `iroh-ctl` with `iroh-one`.

Specifically, `iroh-ctl` is meant to be the communication tool for cloud services. In our last discussion about this, I believe we decided that `iroh-ctl` would stay specifically for coordinating iroh systems that are using multiple processes, and that `iroh-one` would be its own command-line tool, but keep the same commands/api as `iroh-ctl`, just adjusted for the expectation that there is one process.

So in this case, we would instead have a cli binary `iroh-one` that would have some command `iroh-one status -w`

If you have opinions about this, please way in.

**Comment by fabricedesre** [comment]
Ok, I see. To be honest it's a bit surprising because all the current commands from `iroh-ctl` could apply to `iroh-one`. This is a matter of piping all the rpc into a single channel and dispatching internally in the case of `iroh-one`.

No big deal though, because I also need a library version of the rpc to hook up `iroh-one` to other parts of capyloon.  I'll write the command line tool on top of that library instead.

**Comment by ramfox** [comment]
> Ok, I see. To be honest it's a bit surprising because all the current commands from iroh-ctl could apply to iroh-one. This is a matter of piping all the rpc into a single channel and dispatching internally in the case of iroh-one.

oh absolutely. A future refactor we will have an `iroh-api` lib and `iroh-ctl` and `iroh-one` will pull in the same methods to run commands.

So the distinction is more philosophical.

**Comment by b5** [comment]
took us a minute, but we came around to seeing things the same way fabrice does in #314. @fabricedesre, once we've shipped v0.1.0 you'll be able to use the `iroh` CLI to control either either iroh-one or iroh cloud. I'm going to close this issue in favour of tracking progress on #314
